provider/rancher.go

```diff
@@ -320,6 +323,27 @@ func listRancherServices(client *rancher.RancherClient) []*rancher.Service {
 	return servicesList
 }
 
+func listRancherPublicEndponts(client *rancher.RancherClient) map[string]string {
+
+	var publicEndpointMap = make(map[string]string)
```

Comment:
Can be simplified to `publicEndpointMap := make(map[string]string)`.

(Or, even shorter: `publicEndpointMap := map[string]string{}`)